pandas/core/ops.py

```diff
@@ -715,6 +726,29 @@ def wrapper(left, right, name=name, na_op=na_op):
         if isinstance(right, ABCDataFrame):
             return NotImplemented
 
+        if _is_offset(right):
```

Comment:
this logic does not belong here. this is what happens in Timeop.

Reply:
> this logic does not belong here. this is what happens in Timeop.

Yes.  This is preventing this case from being dispatched to DatetimeIndex so that it continues to be handled by TimeOp for now. DatetimeIndex makes a mess of this case (try it...), but I decided to fix that separately.